File: math/primes/is_prime.py


def is_prime(n :int) -> bool :

	if	n <2 : return False
	if	n in ( 2,3 ): return True
	if	not n%2 or not n%3 : return False

	for i in range(5,int(n **.5),6):
		if	not n%i or not n%(i+2): return False
	return	True

File: math/primes/isPrime.java
public Boolean isPrime(int n) {

    if(n <2) return false;
    if(n == 2 || n == 3) return true;
    if(n%2 == 0 || n%3 == 0) return false;

    int i = 5;
    double j = Math.sqrt(n);

    while(i <= j)
        if(n%i == 0 || n%(i+2) == 0) return false;
        else i += 6;
    return true;
}